morse_code.rb

```diff
@@ -0,0 +1,48 @@
+CODE_CHAR = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z'
+}.freeze
+
+def decode_char(morse_code)
+  CODE_CHAR[morse_code] || ''
+end
+
+def decode_word(morse_word)
+  morse_word.split.map { |morse_char| decode_char(morse_char) }.join
+end
+
+def decode_message(morse_message)
+  morse_message.split.map { |morse_word| decode_word(morse_word) }.join
+end
```

Comment:
- [ ] Hey team! 👋 I spotted that there's still an issue with the word spacing. To get the right outcome, consider tweaking your `decode_message` method to look something like this:

```
def decode_message(morse_message)
  morse_words = morse_message.split('   ') # Split by three spaces to separate words
  decoded_words = morse_words.map { |morse_word| decode_word(morse_word) }
  decoded_words.join(' ') # Join the decoded words with a single space
end

```

#### Results

```
"A"
"MY"
"MY NAME"
"A BOX FULL OF RUBIES"

```

#### This should help achieve the desired spacing between words. Keep up the good work! 😊✨